containers/java-postgres/README.md

```diff
@@ -0,0 +1,146 @@
+
+While the definition itself works unmodified, it uses the `mcr.microsoft.com/vscode/devcontainers/java` image which includes `git`, a non-root `vscode` user with `sudo` access, and a set of common dependencies and Java tools for development. You can pick a different version of this image by updating the `VARIANT` arg in `.devcontainer/docker-compose.yml` to pick a Java version.
+
+```yaml
+build:
+  context: ..
+  dockerfile: .devcontainer/Dockerfile
+    args:
+      # Update 'VARIANT' to pick an version of Java: 11, 17.
+      # Append -bullseye or -buster to pin to an OS version.
+      # Use -bullseye variants on local arm64/Apple Silicon.
+      VARIANT: 11-bullseye
+```
+
+You also can connect to PostgreSQL from an external tool when using VS Code by updating `.devcontainer/devcontainer.json` as follows:
+
+```json
+"forwardPorts": [ "5432" ]
+```
+
+### Adding another service
+
+You can add other services to your `docker-compose.yml` file [as described in Docker's documentation](https://docs.docker.com/compose/compose-file/#service-configuration-reference). However, if you want anything running in this service to be available in the container on localhost, or want to forward the service locally, be sure to add this line to the service config:
+
+```yaml
+# Runs the service on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+network_mode: service:db
+```
+
+### Debug Configuration
+
+Note that only the integrated terminal is supported by the Remote - Containers extension. You may need to modify `launch.json` configurations to include the following value if an external console is used.
+
+```json
+"console": "integratedTerminal"
+```
+
+### Installing Maven or Gradle
+
+You can opt to install a version of Maven or Gradle by adding `INSTALL_MAVEN: "true"` or `INSTALL_GRADLE: "true"` to build args in `.devcontainer/docker-compose.yml`. Both of these are set by default. For example:
+
+```yaml
+args:
+  VARIANT: 11
+  INSTALL_GRADLE: "true"
+  INSTALL_MAVEN: "true"
+```
+
+Remove the appropriate arg or set its value to `"false"` to skip installing the specified tool.
+
+You can also specify the version of Gradle or Maven if needed.
+
+```yaml
+args:
+  VARIANT: 11
+  INSTALL_GRADLE: "true"
+  MAVEN_VERSION: "3.8.3"
+  INSTALL_MAVEN: "true"
+  GRADLE_VERSION: "7.2"
+```
+
+### Installing Node.js
+
+Given JavaScript front-end web client code written for use in conjunction with a Java back-end often requires the use of Node.js-based utilities to build, this container also includes `nvm` so that you can easily install Node.js. You can enable installation and change the version of Node.js installed or disable its installation by updating the `args` property in `.devcontainer/docker-compose.yml`.
+
+```yaml
+args:
+  VARIANT: 11
```

Comment:
Wait, what does the variant arg do here in this context?? The node version is below but what's this?

Reply:
I'm going to be honest, I don't know. This part of the README is taken directly from the existing documentation. Let me take a look and see why it's there. @dnastrain any ideas?

Reply:
I found it. It's actually in a comment above this snippet.


      # Update 'VARIANT' to pick an version of Java: 11, 17.
      # Append -bullseye or -buster to pin to an OS version.
      # Use -bullseye variants on local arm64/Apple Silicon.